apps/amo/management/commands/clean_redis.py

```diff
@@ -99,21 +69,47 @@ def file_keys():
             total[1], round(float(total[1]) / total[0] * 100, 1)))
 
 
+def get_redis_backend(backend_uri):
+    # From django-redis-cache
+    # This is temporary https://github.com/washort/nuggets/pull/1
+    _, server, params = parse_backend_uri(backend_uri)
+    db = params.pop('db', 1)
+    try:
+        db = int(db)
+    except (ValueError, TypeError):
+        db = 0
+    try:
+        socket_timeout = float(params.pop('socket_timeout'))
+    except (KeyError, ValueError):
+        socket_timeout = None
+    password = params.pop('password', None)
+    if ':' in server:
+        host, port = server.split(':')
+        try:
+            port = int(port)
+        except (ValueError, TypeError):
+            port = 6379
+    else:
+        host = 'localhost'
+        port = 6379
+    return redislib.Redis(host=host, port=port, db=db, password=password,
+                          socket_timeout=socket_timeout)
+
+
 class Command(BaseCommand):
     help = "Clean up the redis used by cache machine."
 
     def handle(self, *args, **kw):
         try:
-            master = redisutils.connections['cache']
-            slave = redisutils.connections['cache_slave']
+            master = get_redis_backend(settings.REDIS_BACKENDS['cache'])
+            slave = get_redis_backend(settings.REDIS_BACKENDS['cache_slave'])
+            # master = redisutils.connections['cache']
+            # slave = redisutils.connections['cache_slave']
         except Exception:
             log.error('Could not connect to redis.', exc_info=True)
             return
-        if args:
-            filename = args[0]
-            try:
-                cleanup(master, slave, filename)
-            finally:
-                os.unlink(filename)
-        else:
-            vacuum(master, slave)
+
+        try:
+            cleanup(master, slave)
+        except Exception:
+            log.error('Cleanup failed.', exc_info=True)
```

Comment:
this should end in a `raise`. Again, after that change you wouldn't need `exc_info` in the logging. Without raise, the script will exit zero which seems like not what we want.